digi/xbee/packets/socket.py

```diff
@@ -0,0 +1,2932 @@
+                else utils.hex_to_string(self.__dest_address)}
+
+    def __get_socket_id(self):
+        """
+        Returns the the socket ID.
+
+        Returns:
+            Integer: the socket ID.
+        """
+        return self.__socket_id
+
+    def __set_socket_id(self, socket_id):
+        """
+        Sets the socket ID.
+
+        Args:
+            socket_id (Integer): the new socket ID.
+
+        Raises:
+            ValueError: if ``socket_id`` is less than 0 or greater than 255.
+        """
+        if socket_id < 0 or socket_id > 255:
+            raise ValueError("Socket ID must be between 0 and 255")
+        self.__socket_id = socket_id
+
+    def __get_dest_port(self):
+        """
+        Returns the destination port.
+
+        Returns:
+            Integer: the destination port.
+        """
+        return self.__dest_port
+
+    def __set_dest_port(self, dest_port):
+        """
+        Sets the destination port.
+
+        Args:
+            dest_port (Integer): the new destination port.
+
+        Raises:
+            ValueError: if ``dest_port`` is less than 0 or greater than 65535.
+        """
+        if dest_port < 0 or dest_port > 65535:
+            raise ValueError("Destination port must be between 0 and 65535")
+        self.__dest_port = dest_port
+
+    def __get_dest_address_type(self):
+        """
+        Returns the destination address type.
+
+        Returns:
+            Integer: the destination address type.
+        """
+        return self.__dest_address_type
+
+    def __set_dest_address_type(self, dest_address_type):
+        """
+        Sets the destination address type.
+
+        Args:
+            dest_address_type (Integer): the new destination address type.
+
+        Raises:
+            ValueError: if ``dest_address_type`` is different than :attr:`SocketConnectPacket.DEST_ADDRESS_BINARY` and
+                        :attr:`SocketConnectPacket.DEST_ADDRESS_STRING`.
+        """
+        if dest_address_type not in (SocketConnectPacket.DEST_ADDRESS_BINARY, SocketConnectPacket.DEST_ADDRESS_STRING):
+            raise ValueError("Destination address type must be %d or %d" % (SocketConnectPacket.DEST_ADDRESS_BINARY,
+                                                                            SocketConnectPacket.DEST_ADDRESS_STRING))
+        self.__dest_address_type = dest_address_type
+
+    def __get_dest_address(self):
+        """
+        Returns the destination address.
+
+        Returns:
+            Bytearray or String: the destination address.
+        """
+        return self.__dest_address
+
+    def __set_dest_address(self, dest_address):
+        """
+        Sets the destination address.
+
+        Args:
+            dest_address (Bytearray or String): the new destination address.
+
+        Raises:
+            ValueError: if ``dest_address`` is ``None`` or does not follow the format specified in the configured type.
```

Comment:
How are exceptions supposed to be described? I have seen that in some occasions, we document different causes of the same exception separately, but other times, like this one, thetwo possible causes of the exception are documented together.
How should this be approached? in any case, either this is the correct form or not, all exceptions should be documented in the same manner.